cunumeric/array.py

```diff
@@ -2994,61 +2993,41 @@ def reshape(self, shape, order="C") -> ndarray:
         --------
         Multiple GPUs, Multiple CPUs
         """
-        if shape != -1:
-            # Check that these sizes are compatible
-            if isinstance(shape, Iterable):
-                newsize = 1
-                newshape = list()
-                unknown_axis = -1
-                for ax, dim in enumerate(shape):
-                    if dim < 0:
-                        newshape.append(np.newaxis)
-                        if unknown_axis == -1:
-                            unknown_axis = ax
-                        else:
-                            unknown_axis = -2
-                    else:
-                        newsize *= dim
-                        newshape.append(dim)
-                if unknown_axis == -2:
-                    raise ValueError("can only specify one unknown dimension")
-                if unknown_axis >= 0:
-                    if self.size % newsize != 0:
-                        raise ValueError(
-                            "cannot reshape array of size "
-                            + str(self.size)
-                            + " into shape "
-                            + str(tuple(newshape))
-                        )
-                    newshape[unknown_axis] = self.size // newsize
-                    newsize *= newshape[unknown_axis]
-                if newsize != self.size:
-                    raise ValueError(
-                        "cannot reshape array of size "
-                        + str(self.size)
-                        + " into shape "
-                        + str(shape)
-                    )
-                shape = tuple(newshape)
-            elif isinstance(shape, int):
-                if shape != self.size:
-                    raise ValueError(
-                        "cannot reshape array of size "
-                        + str(self.size)
-                        + " into shape "
-                        + str((shape,))
-                    )
-            else:
-                TypeError("shape must be int-like or tuple-like")
+
+        if len(args) == 0:
+            raise TypeError("reshape() takes exactly 1 argument (0 given)")
+        elif len(args) == 1:
+            shape = (args[0],) if isinstance(args[0], int) else args[0]
         else:
-            # Compute a flattened version of the shape
-            shape = (self.size,)
+            shape = args
+
+        computed_shape = tuple(operator.index(extent) for extent in shape)
```

Comment:
Why this? Probably doesn't hurt, just curious about when this might come up.

Reply:
that line is exactly for the case where someone tries `arr.reshape(ext1, ext2, ext3, ...)`, which fixes #127.

Reply:
Asking specifically about the `operator.index(.)` call

Reply:
that code window wasn't showing the line you were referring to. someone might try to do something like `arr.reshape(1, (2,3))`